### PR TITLE
Added a driver for Canon cameras. The specifications of the sensor an…

### PR DESCRIPTION
**Pull request by wvreeven**
…d ISO are for a Canon 5D Mark IV.

### CODE REVIEW

File: python/lsst/ts/GenericCamera/driver/canoncamera.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        await super().startIntegration()
+
+    async def endReadout(self):
+        """End reading out the image.
+        """
+        # The image can be obtained by reading out the file_path variable in
+        # which the path to the file on the camera was stored.
+        target = os.path.join("/tmp", self.file_path.name)
+        camera_file = self.camera.file_get(self.file_path.folder, self.file_path.name, gp.GP_FILE_TYPE_NORMAL)
+        camera_file.save(target)
```
Comment (tribeiro) [inline python/lsst/ts/GenericCamera/driver/canoncamera.py:134]: Do your really need to store it in a file to be able to read it? what kind of object is `camera_file`?

File: schema/GenericCamera.yaml [hunk 2]
```diff
@@ -102,4 +103,20 @@ allOf:
...
+        default: 0
+        type: number
+      binValue:
+        default: 1
+        type: number
+      ImageType:
+        default: RAW
+        type: string
+        enum:
+          - RAW
```
Comment (tribeiro) [inline schema/GenericCamera.yaml:122]: If you can't read the ROI from the camera, you may want to add it as a configuration parameter.

File: python/lsst/ts/GenericCamera/driver/canoncamera.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        -------
+        int
+            The top most pixel of the region.
+        int
+            The left most pixel of the region.
+        int
+            The width of the region in pixels.
+        int
+            The height of the region in pixels."""
+        return 0, 0, 6720, 4480
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/GenericCamera/driver/canoncamera.py:88]: Yes it is for the Canon EOS 5D Mark IV camera. Unfortunately Canon has not made their API available so the camera cannot provide the sensor dimensions via libgphoto2. I'll add the ROI as a configuration parameter as you suggested in your other comment.

**Comment by tribeiro** [comment]
So, the unit tests are failing because there are some library missing. Can you add a step in the Jenkinsfile that installs the missing library? I would suggest adding a `requirements.txt` and doing `pip install -r requirements.txt`.

I can add them to the develop-env later but it might be a good idea to try to capture these new dependencies.

**Comment by wvreeven** [comment]
> So, the unit tests are failing because there are some library missing. Can you add a step in the Jenkinsfile that installs the missing library? I would suggest adding a `requirements.txt` and doing `pip install -r requirements.txt`.
> 
> I can add them to the develop-env later but it might be a good idea to try to capture these new dependencies.

Yes I will do that. Unfortunately, to be able to install the gphoto python library, libgphoto2-devel needs to have been installed in the environment otherwise the installation will fail. I am afraid that this will not be resolved until the docker image contains that development library.

**Comment by wvreeven** [comment]
Changes pushed. Can you have another look please?

**Comment by wvreeven** [comment]
I finally managed to get the build pass successfully. I had to switch to my own docker image and disable the Python API reference doc generation. Both are temporary measures. Unfortunately I don't understand why generating the docs is failing but it seems that Sphinx doesn't handle well the imports of the ctypes in andorcamera.py and zwocamera.py. I'll investigate further but in the mean time would appreciate it if you could approve of my code changes so at least I can continue with the Canon driver. Thanks!

**Pull request merged**